app/views/play/OzariaUnitMap.vue

```diff
@@ -56,7 +58,8 @@ export default Vue.extend({
 
     try {
       this.levelSessions = await api.users.getLevelSessions({userID: me.get('_id')})
```

Comment:
I dont think the props for the unit map would/should change from somewhere, however its a good idea to create a separate function for loading data and calling it from created. Not sure if an updated method is required for now.

Reply:
How about when the slug changes for 1FH and 1UP?

Reply:
I think that should call the `created` function, but not entirely sure. Do you know, if we click on a link from another page that opens this page with a different campaign name - would that call the `created` function (asking in context of the using vue router for all the pages, i.e. being part of the single vue app)?

Reply:
The short answer is not always.  This is highly dependent on implementation and the specific scenario, which is why its good go handle both the created / updated scenario like we did with the nested intro level components.  The assumption here is that when the URL changes the page will be rendered.  

This works most of the time but can behave strangely in certain scenarios.  One of those scenarios is navigating from `/play/intro/1fh` to `/play/intro/1up`.  In this scenario Vue Router will update the component tree and update the props associated with the component tree (changing the intro level slug from `1fh` to `1up`) but it will not unmount and remount the components.  Vue does this to optimize render performance.

More generally any time that this component is updated without being added or removed we'll run into strange edge cases where this is simply not rendering.  To prevent these issues in the future we should add the updated callback now.  There are likely quite a few other components we don't handle this in and we should keep an eye out for these as we move forward too.

Reply:
Thanks for the explanations @jmif, this was helpful!
I will add `updated` here and keep an eye out for future vue components as well so that the scenario when it is not re-rendered gets handled.

Reply:
@jmif Looks like `updated` is called after DOM is re-rendered and it is not recommended to change state in this hook. Using watchers instead! Reference: https://vuejs.org/v2/api/#updated

Although it doesn't look scalable if there are a lot of props to watch for.